### PR TITLE
Bug 1190790 - Make sure that panels are populated correctly when navigating to them r=drs

### PR DESCRIPTION
**Pull request by gabrielesvelto**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e64bbd7a58831dfb65ab48e089ef484583f30cbc

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=34072d8c7624af27e2aa33841163c8072bf9ae60

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5c44acaaa624e98bfbcf047c475aaa30c1ea55de

File: apps/communications/dialer/index.html [hunk 1]
```diff
@@ -46,16 +47,24 @@
...
-    <script defer src="/shared/js/dialer/keypad.js"></script>
     <script defer src="/shared/js/dialer/font_size_manager.js"></script>
+    <script defer src="/shared/js/dialer/keypad.js"></script>
+    <script defer src="/shared/js/dialer/tone_player.js"></script>
+    <script defer src="/shared/js/dialer/utils.js"></script>
+    <script defer src="/shared/js/dialer/voicemail.js"></script>
 
     <script defer src="/dialer/js/navigation.js"></script>
     <script defer src="/dialer/js/dialer.js"></script>
+    <script defer src="/dialer/js/call_log.js"></script>
```
Comment (DouglasSherk) [inline apps/communications/dialer/index.html]: How did this happen? How was it getting loaded before?

File: apps/communications/dialer/js/dialer.js [hunk 2]
```diff
@@ -1,16 +1,14 @@
...
+          MmiManager, Notification, NotificationHelper, SettingsListener,
+          SimSettingsHelper, SuggestionBar, TelephonyHelper, Utils, Voicemail,
+          MozActivity, Navigation */
 
 var NavbarManager = {
   init: function nm_init() {
-    this.update = this.update.bind(this);
-    this.update('option-keypad');
-    
+    Navigation.show('keypad');
```
Comment (DouglasSherk) [inline apps/communications/dialer/js/dialer.js]: This should be tested. It's pretty critical.

File: apps/communications/dialer/index.html (hunk 1 above)
Comment (gabrielesvelto) [inline apps/communications/dialer/index.html]: This was loaded by ``Navbar.ensureResources()``, see here:

https://github.com/mozilla-b2g/gaia/pull/31641/files#diff-660f9c7d029624283424917265eb27f5L46

As with the other bits these were lazy-loaded in the startup path so I just moved them to be greedy loaded instead.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=872025ebc7bda46bb541e79855fdfc6b0acedb83

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=17d4b2434e21c6f79252a2d207c70bc7329cebd7

**Pull request merged**